src/rdpc/variantCallingAnalysesProcessor.ts

```diff
@@ -98,23 +98,44 @@ export const getEarliestDateForDonor = (
   Object.entries(donors).forEach(([donorId, info]) => {
     const wfInfo: WorkflowInfo = { sangerVC: [], mutect: [], openAccess: [] };
 
+    // Skip the current analysis if its firstPublishedAt is null or empty.
     const earliestSanger = _.head(
-      _.sortBy(info.sangerVC, (sanger) => Number(sanger.firstPublishedAt))
+      _.sortBy(
+        _.filter(
+          info.sangerVC,
+          (sanger) =>
+            sanger.firstPublishedAt !== null && sanger.firstPublishedAt !== ""
+        ),
+        (sanger) => Number(sanger.firstPublishedAt)
+      )
     );
```

Comment:
this `_.head(_.sortBy(_.filter(...)...)...)` is repeated 3x. do you want to make a single function athat takes a list of analyses and applies these transformations?

Reply:
function added

Reply:
nice